lib/bedrock.js

```diff
@@ -0,0 +1,58 @@
+class DataProvider {
+  constructor (indexes) {
+    this.data = indexes
+    this.blockPalette = []
+    this.blockPaletteNamed = {}
+    this.itemPalette = {}
+    this.itemPaletteNamed = {}
+  }
+
+  /**
+   * Loads a block palette from blockStates.json. This data used to be sent over the network, it's now hard-coded.
+   * "loadBlocks" loads instead from blocks.json.
+   * The index to a block in this array is its "Runtime ID".
+   * @param {Array<object>} entries
+   */
+  loadBlockPalette (entries) {
+    for (let i = 0; i < entries.length; i++) {
+      const entry = entries[i]
+      // console.log(entry)
+      entry.runtimeId = i
+      this.blockPalette.push(entry)
+      this.blockPaletteNamed[entry.name] = entry
+    }
+  }
+
+  /**
+   * Loads an item palette, sent by the server
+   * @param {{ name, runtimeId, componentBased }} entries
+   */
+  loadItemPalette (entries) {
```

Comment:
This is a design change for node-minecraft-data as it would now accept data and not only provide data.
I am wondering as to whether this should go to mineflayer instead
But there are advantages to putting it there
At least it should be documented

Reply:
So yes I think there needs to be some more thought into the best approach here. Inventory code specifically would depend on this, and it would mean that the bulk of the inventory code would have to go into mineflayer, or go through some hoops there. It's not ideal to put it here I agree, but maybe some other package.